### DISCUSSION TITLE
Can you inject custom values into the method signature through a custom decorator if its an application command?

### DISCUSSION THREAD

**Discussion by Zitronenjoghurt** [Rapptz/discord.py#9727] [Q&A]
In my custom decorator I am trying to pass a refund function to the greet method so that in case of an error the costs of the command can easily be refunded:
```py
@relationship_group.command(name="greet", description="Greet someone c:")
@app_commands.describe(member="The user you want to greet")
@command_cost(amount=10, command_name="greet")
async def greet(self, interaction: discord.Interaction, member: discord.Member):
    ...
```
Though as I have noticed all parameters in that method except interaction are automatically turned into command parameters. Is there some way around that? I saw that some decorators have an extras field, it would be really helpful to have something like that for the method signature too.

**Comment by No767** [comment]
@Zitronenjoghurt the simple answer is no. Application commands cannot be called by other application commands, nor would I be in support of an implementation as you described. In your case, it is advisable to separate the logic from your refund command into an function, and reuse it in your greet command.

**Reply by Zitronenjoghurt** [reply]
Yeah it is alright to build around that, itd just have been more convenient :o I really like decoraters and using them with my commands.
